vars/buildPlugin.groovy

```diff
@@ -123,23 +123,31 @@ def call(Map params = [:]) {
                   }
                   mavenOptions += 'clean install'
                   try {
+                    boolean usingArtifactCachingProxy = false
                     if (artifactCachingProxyEnabled) {
-                      // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
-                      final String defaultProxyProvider = 'azure'
-                      def availableProxyProviders = ['azure', 'do']
-                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
-                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
-                        requestedProvider = defaultProxyProvider
-                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      // As the env var ARTIFACT_CACHING_PROXY_PROVIDER can't be set on Azure VM agents,
+                      // we're using 'azure' as default provider if none is specified
+                      final String requestedProxyProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER ?: 'azure'
+                      final String validProxyProviders = ['aws', 'azure', 'do']
+                      final String configuredAvailableProxyProviders = env.ARTIFACT_CACHING_PROXY_AVAILABLE_PROVIDERS
+                      if (configuredAvailableProxyProviders != null && configuredAvailableProxyProviders != '') {
+                        final String availableProxyProviders = configuredAvailableProxyProviders.split(',')
+                        // Configure Maven settings if the requested provider is valid and available
+                        if (validProxyProviders.contains(requestedProxyProvider) && availableProxyProviders.contains(requestedProxyProvider)) {
```

Comment:
Note: with this structure, the next step for having a fallback if the provider doesn't respond to an health check request is simple:

```suggestion
                        // Configure Maven settings if the requested provider is valid, available and respond to an health check
                        if (validProxyProviders.contains(requestedProxyProvider)
                            && availableProxyProviders.contains(requestedProxyProvider)
                            && sh(script: "curl --fail https://repo.${requestedProxyProvider}.jenkins.io/healthz", returnStatus: true) == 0
                            ) {
```
I'll open a PR as soon as this one is merged.